/*
 * This file was automatically generated by EvoSuite
 * Thu Jun 08 12:55:38 GMT 2017
 */

package tutorial;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import java.util.NoSuchElementException;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;
import tutorial.LinkedListIterator;
import tutorial.Node;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true) 
public class LinkedListIterator_ESTest extends LinkedListIterator_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      Node<Object> node0 = new Node<Object>((Object) null);
      LinkedListIterator<Object> linkedListIterator0 = new LinkedListIterator<Object>(node0);
      linkedListIterator0.next();
      // Undeclared exception!
      try { 
        linkedListIterator0.next();
        fail("Expecting exception: NoSuchElementException");
      
      } catch(NoSuchElementException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("tutorial.LinkedListIterator", e);
      }
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      Object object0 = new Object();
      Node<Object> node0 = new Node<Object>(object0);
      LinkedListIterator<Object> linkedListIterator0 = new LinkedListIterator<Object>(node0);
      Object object1 = linkedListIterator0.next();
      assertSame(object0, object1);
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      Node<String> node0 = new Node<String>("1oP3=8sTd1JMsSdA");
      LinkedListIterator<String> linkedListIterator0 = new LinkedListIterator<String>(node0);
      boolean boolean0 = linkedListIterator0.hasNext();
      assertTrue(boolean0);
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      LinkedListIterator<Object> linkedListIterator0 = new LinkedListIterator<Object>((Node<Object>) null);
      boolean boolean0 = linkedListIterator0.hasNext();
      assertFalse(boolean0);
  }

  @Test(timeout = 4000)
  public void test4()  throws Throwable  {
      LinkedListIterator<Object> linkedListIterator0 = new LinkedListIterator<Object>((Node<Object>) null);
      // Undeclared exception!
      try { 
        linkedListIterator0.remove();
        fail("Expecting exception: UnsupportedOperationException");
      
      } catch(UnsupportedOperationException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         assertThrownBy("tutorial.LinkedListIterator", e);
      }
  }
}
